### PR TITLE
modify comments to enable javadoc

### PR DESCRIPTION
**Pull request by abekornelis**
I copied over all javadoc-enabling changes that Hugh applied. 
Additionally, I modified all routine headers in vz390.java to enable javadoc
And I modified bldjar.bat to generate javavdoc before compiling z390

### CODE REVIEW

File: src/ez390.java [hunk 1]
```diff
@@ -246,39 +247,55 @@
...
-private void run_pgm(int zcvt_pgm_addr){
-	/*
+
+
+
+	/**
 	 * execute IPL pgm and/or application pgm
 	 */
+	 
+private void run_pgm(int zcvt_pgm_addr){
```
Comment (jyganci) [inline src/ez390.java:298]: Not sure whether one puts javadoc-style comments for a private method. If you do, add "@param zcvt_pgm_addr" in the above javadoc.

File: src/ez390.java [hunk 2]
```diff
@@ -344,7 +370,10 @@ private void init_ez390(String[] args, JTextArea log_text, JTextField command_te
 	 *     timeout, and cpu rate statistics
 	 * 
 	 * 
+     *     </pre>
 	 */
+
+private void init_ez390(String[] args, JTextArea log_text, JTextField command_text){
```
Comment (jyganci) [inline src/ez390.java:376]: Add 3 @param lines in the above javadoc.

File: src/gz390.java [hunk 3]
```diff
@@ -3148,7 +3148,7 @@ private String get_ascii_string(byte[] text_byte,int lbuff){
 	}
 	return text;
 }
-     /***********************************************
+     /* *********************************************
```
Comment (jyganci) [inline src/gz390.java:3151]: If this is supposed to be javadoc, shouldn't it be "/** ********************************"??? That is, "/**". followed by one space, followed by rest of asterisks?

File: src/sz390.java [hunk 4]
```diff
@@ -1020,10 +1033,14 @@ public synchronized void put_log(String msg) {
...
-	   /*
-	    * put line to listing file
-	    */
+
+
+
+   /**
+    * put line to listing file
+    */
+    private void put_log_line(String msg){
```
Comment (jyganci) [inline src/sz390.java:1042]: Add @param line to javadoc.

File: src/sz390.java [hunk 5]
```diff
@@ -1040,22 +1057,31 @@ private void put_log_line(String msg){
 	   		   }
 	   	   }
 	   }
+
+
+
+    /**
+     * put msg to console or cmd process output
+     * and yield to let parent process m
+     */
```
Comment (jyganci) [inline src/sz390.java:1066]: Add @param line to javadoc.

File: src/tz390.java [hunk 6]
```diff
@@ -4531,11 +4577,15 @@ else if (opt_machine.equals("Z15")) // rpi 2202
...
-private void process_option(String opt_file_name,int opt_file_line,String token){
-	/*
+
+
+
+	/**
 	 * process option from command line or
 	 * from @file optionsfile line.
 	 */
+
```
Comment (jyganci) [inline src/tz390.java:4587]: Add 3 @param to javadoc.

File: src/tz390.java [hunk 7]
```diff
@@ -5297,21 +5352,31 @@ public  void set_trace_options(String trace_options){
    		index++;
    	}
 }
+
+
+
+    /**
+     * collect invalid options for single error
+     */
```
Comment (jyganci) [inline src/tz390.java:5360]: Add @param lines to javadoc.

File: src/tz390.java [hunk 8]
```diff
@@ -5297,21 +5352,31 @@ public  void set_trace_options(String trace_options){
...
+
+    /**
+     * process option file as follows:
+     * <ol>
+     * <li> Default suffix .OPT </li>
+     * <li> Uses SYSOPT path which defaults to program path </li>
+     * <li> Comments starting with * to end of line </li>
+     * <li> @file option can be nested. </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/tz390.java:5377]: Add @param lines to javadoc.

File: src/tz390.java [hunk 9]
```diff
@@ -5515,22 +5608,32 @@ public void close_trace_file(){
...
     return (int)(mem_tot >> 20);
 }
+
+
+
+    /**
+     * return shortest file name possible
+     * with quotes if LSN
+     * @param file_name - name of the file
+     */
```
Comment (jyganci) [inline src/tz390.java:5634]: Add @return to javadoc.

File: src/tz390.java [hunk 10]
```diff
@@ -6649,11 +6987,15 @@ public boolean fp_get_dfp_bin(int dfp_type,BigDecimal dfp_bd){
...
-    	 * truples of 10 bits representing 3 digits.
-    	 */
+
+
+
+    /**
+     * return long with 1 to 6 DPD densly packed deciaml
+     * truples of 10 bits representing 3 digits.
+     */
+private long get_dfp_ccf_digits(int tot_digits,int digit_offset, int digit_count){
```
Comment (jyganci) [inline src/tz390.java:6997]: Add @param and @return values to javadoc.

File: src/tz390.java [hunk 11]
```diff
@@ -6814,22 +7208,30 @@ public void sleep_now(long mills){
...
-		 *     else replace existing path option.
-		 */
+
+
+
+    /**
+     * 1.  if new path starts with +, concat with existing path
+     *     else replace existing path option.
+     */
+private String set_path_option(String old_path,String new_path){
```
Comment (jyganci) [inline src/tz390.java:7218]: Add @param and @return values to javadoc.

File: src/tz390.java [hunk 12]
```diff
@@ -7234,11 +7636,15 @@ public void put_stat_final_options(){
...
-		 * add final option value to final_option
-		 * formatted string.
-		 */
+
+
+
+    /**
+     * add final option value to final_option
+     * formatted string.
+     */
```
Comment (jyganci) [inline src/tz390.java:7645]: Add @param to javadoc.

File: src/tz390.java [hunk 13]
```diff
@@ -7469,10 +7908,14 @@ private void list_ebcdic_ascii_unicode(){
...
-   	 private String map_text(int index){
-   		 /*
-   		  * return text index,hex-ebcdic,hex-ascii,char,U+hex
-   		  */
+
+
+
+    /**
+     * return text index,hex-ebcdic,hex-ascii,char,U+hex
+     */
```
Comment (jyganci) [inline src/tz390.java:7916]: Add @return to javadoc.

File: src/tz390.java [hunk 14]
```diff
@@ -7493,7 +7936,9 @@ private String map_text(int index){
    	     return text;
 	}
 
-	private void list_available_charsets(){
+
+
+private void list_available_charsets(){
```
Comment (jyganci) [inline src/tz390.java:7941]: No javadoc???

File: src/tz390.java [hunk 15]
```diff
@@ -7521,54 +7966,66 @@ private void list_available_charsets(){
...
-		 * return true if test_ebcdic charset
-		 * meets minimum character requirements
-		 */
+
+
+
+    /**
+     * return true if test_ebcdic charset
+     * meets minimum character requirements
+     */
```
Comment (jyganci) [inline src/tz390.java:8026]: Add @return to javadoc.

File: src/tz390.java [hunk 16]
```diff
@@ -7652,11 +8112,15 @@ public void list_hex_ascii_ebcdic(){
...
-		 * load ebcdic_charset_name as alternate
-		 * source for system defiend ebcdic_charset_name
-		 */
+
+
+
+    /**
+     * load ebcdic_charset_name as alternate
+     * source for system defiend ebcdic_charset_name
+     */
```
Comment (jyganci) [inline src/tz390.java:8121]: Add @return to javadoc.

File: src/vz390.java [hunk 17]
```diff
@@ -986,12 +1009,17 @@ public void svc_close_acb() {
...
 
+
+
+    /**
+     * load VCDT using ACB DSNAME or DDNAME
+     * <ol>
+     * <li> If file spec includes dot, use suffix to find VCDT entry else use ACBNAME field. </li> 
+     * <li> Set ACBDCBN, and ACBDCBA from VCDT VCLR or VPTH entry. </li> 
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:1020]: Add @return to javadoc.

File: src/vz390.java [hunk 18]
```diff
@@ -1014,13 +1042,22 @@ private boolean load_vcdt() {
...
+     * <ol>
+     * <li> Set cur_vclra_addr </li> 
+     * <li> Set cur_vptha_addr or 0 </li> 
+     *    <ol>
+     *    <li> If vpth_flag_aixp, set acb_oflgs_aixp else 0 </li> 
+     *    <li> If vpth_flag_aixu, set acb_oflgs_aixu else 0 </li> 
+     *    </ol> </li>
+     * <li> Set cur_vcdt_dcba for use by init_acb_dcb. </li> 
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:1058]: Add @return to javadoc.

File: src/vz390.java [hunk 19]
```diff
@@ -1188,11 +1237,16 @@ private boolean check_acb_macrf() {
...
-		 * upgrade VXN's. 2. Alloc memory for last key if KSDS.
-		 */
+
+
+    /**
+     * <ol>
+     * <li> dynamically allocate and open dcbs required for VES, VX0, and any upgrade VXN's. </li> 
+     * <li> Alloc memory for last key if KSDS. </li> 
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:1247]: Add @return to javadoc.

File: src/vz390.java [hunk 20]
```diff
@@ -1252,12 +1306,14 @@ private boolean open_acb_dcbs() {
...
-		/*
-		 * copy model dcb from vcdt_dcba to new dynamcially allocated dcb
-		 * address and set DCBLRECLF and DCBDSNAM fields
-		 */
+
+
+    /**
+     * copy model dcb from vcdt_dcba to new dynamcially allocated dcb
+     * address and set DCBLRECLF and DCBDSNAM fields
+     */
```
Comment (jyganci) [inline src/vz390.java:1314]: Add @param values to javadoc.

File: src/vz390.java [hunk 21]
```diff
@@ -1268,10 +1324,13 @@ private void init_acb_dcb(int dcb_addr, int dcb_lrecl_f, int dcb_dsname,
...
 
-	private boolean open_acb_dcb(int dcb_addr) {
-		/*
-		 * dynamically allocate and open DCB for VES, VX0, VXN's
-		 */
+
+
+    /**
+     * dynamically allocate and open DCB for VES, VX0, VXN's
+     */
```
Comment (jyganci) [inline src/vz390.java:1331]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 22]
```diff
@@ -1287,10 +1346,13 @@ private boolean open_acb_dcb(int dcb_addr) {
...
 
-	private boolean close_acb_dcbs() {
-		/*
-		 * close dynamically allocated ACB DCB's
-		 */
+
+
+    /**
+     * close dynamically allocated ACB DCB's
+     */
```
Comment (jyganci) [inline src/vz390.java:1353]: Add @return to javadoc.

File: src/vz390.java [hunk 23]
```diff
@@ -1457,26 +1531,42 @@ private void rpl_get_esds_adr() {
...
-		 */
+
+
+    /**
+     * <ol>
+     * set RPLLXRBA last rec XRBA
+     * <li> VES for ESDS/RRDS </li>
+     * <li> VX0 for KSDS/VRRDS </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:1542]: Add @param to javadoc.

File: src/vz390.java [hunk 24]
```diff
@@ -1457,26 +1531,42 @@ private void rpl_get_esds_adr() {
...
 
-	private void set_rpl_cxrba(long xrba) {
-		/*
-		 * set RPL cur pos xrba 1. VES for ESDS/RRDS 2. VX0 for KSDS/VRRDS
-		 */
+
+
+    /**
+     * set RPL cur pos xrba 1. VES for ESDS/RRDS 2. VX0 for KSDS/VRRDS
+     */
```
Comment (jyganci) [inline src/vz390.java:1552]: Add @param to javadoc.

File: src/vz390.java [hunk 25]
```diff
@@ -1554,10 +1650,13 @@ private void rpl_get_ksds_seq() {
...
 
-	private boolean get_first_seq_ksir() {
-		/*
-		 * position to first KSIR in KSIT for either FWD or BWD seq. access.
-		 */
+
+
+    /**
+     * position to first KSIR in KSIT for either FWD or BWD seq. access.
+     */
```
Comment (jyganci) [inline src/vz390.java:1657]: Add @return to javadoc.

File: src/vz390.java [hunk 26]
```diff
@@ -1569,11 +1668,14 @@ private boolean get_first_seq_ksir() {
...
-		/*
-		 * read record from current KSIR and postion to next KSIR or index entry
-		 * or return eod
-		 */
+
+
+    /**
+     * read record from current KSIR and postion to next KSIR or index entry
+     * or return eod
+     */
```
Comment (jyganci) [inline src/vz390.java:1676]: Add @return to javadoc.

File: src/vz390.java [hunk 27]
```diff
@@ -1598,22 +1700,27 @@ private boolean read_ksir_cur_rec() {
...
-		 * display error message for KSIR 
-		 * broken links and set feedback ves data error
-		 */
+
+
+
+    /*
+     * display error message for KSIR 
+     * broken links and set feedback ves data error
+     */
```
Comment (jyganci) [inline src/vz390.java:1709]: Add @param values to javadoc.

File: src/vz390.java [hunk 28]
```diff
@@ -1598,22 +1700,27 @@ private boolean read_ksir_cur_rec() {
...
-		/*
-		 * set cur_vx0_xrba to next ksds index going forward or backward else
-		 * set eod
-		 */
+
+
+    /**
+     * set cur_vx0_xrba to next ksds index going forward or backward else
+     * set eod
+     */
```
Comment (jyganci) [inline src/vz390.java:1722]: Add @return to javadoc.

File: src/vz390.java [hunk 29]
```diff
@@ -1787,11 +1918,14 @@ private void rpl_get_rrds_seq() {
...
-		/*
-		 * search vx0 for ksds key in rplarg and set cur_vx0_xrba entry if found
-		 * else false.
-		 */
+
+
+    /**
+     * search vx0 for ksds key in rplarg and set cur_vx0_xrba entry if found
+     * else false.
+     */
```
Comment (jyganci) [inline src/vz390.java:1926]: Add @return to javadoc.

File: src/vz390.java [hunk 30]
```diff
@@ -1899,11 +2033,14 @@ private boolean find_ksds_key() {
...
-		/*
-		 * set rpl_flag_getok or rpl_flag_getnf and save along with
-		 * rpl_flag_ksit possible following put or insert.
-		 */
+
+
+    /**
+     * set rpl_flag_getok or rpl_flag_getnf and save along with
+     * rpl_flag_ksit possible following put or insert.
+     */
```
Comment (jyganci) [inline src/vz390.java:2041]: Add @param to javadoc.

File: src/vz390.java [hunk 31]
```diff
@@ -1914,26 +2051,30 @@ private void set_rpl_cur_rec(boolean found) {
...
+
+    /**
+     * search ksit binary tree at cur_vx0_xrba (negative) 
+     * <ol>
+     * <li> Set cur_ves_xrba to record if found </li>
+     * <li> Set cur_rpl_ksir to last ksir for use by insert_ksir </li>
+     * <li> Set avl_r_xrba/par/low/high if ksir found requiring AVL
+     *    rotation to keep tree balanced after KSIR is inserted. </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:2076]: Add @return to javadoc.

File: src/vz390.java [hunk 32]
```diff
@@ -1989,10 +2130,13 @@ private boolean find_ksir() {
...
 
-	private boolean get_ksir_rec() {
-		/*
-		 * set cur_ves_xrba to ksir rec unless deleted
-		 */
+
+
+    /**
+     * set cur_ves_xrba to ksir rec unless deleted
+     */
```
Comment (jyganci) [inline src/vz390.java:2137]: Add @return to javadoc.

File: src/vz390.java [hunk 33]
```diff
@@ -2002,15 +2146,25 @@ private boolean get_ksir_rec() {
...
+      *    <li> -1 if 0 return false (no rec found) </li>
+      *    <li> xrba and return true. </li> </li>
+      *    </ol> </li>
+      * </ol>
+      * Note:
+      * <ol>
+      * <li> VX0 XRBA's for valid records are stored +1. to distinguish 0 as unwritten VES XRBA. </li>
+      * </ol>
+      */
+    private boolean get_rrds_ves_xrba(boolean key) {
```
Comment (jyganci) [inline src/vz390.java:2166]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 34]
```diff
@@ -2238,12 +2407,13 @@ private void rpl_put_rrds() {
...
-		/*
-		 * rewrite RRDS or KSDS record at cur_ves_xrba and update index xrba if
-		 * it changed due to variable length change.
-		 * 
-		 */
+    /**
+     * rewrite RRDS or KSDS record at cur_ves_xrba and update index xrba if
+     * it changed due to variable length change.
+     * 
+     */
```
Comment (jyganci) [inline src/vz390.java:2414]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 35]
```diff
@@ -2277,11 +2447,14 @@ private boolean rewrite_ves_rec(int tiot_index, long xrba_index) {
...
-		/*
-		 * add RRDS or KSDS record to VES and update index XRBA address which
-		 * may be in VX0 or in VES KSIR.
-		 */
+
+
+    /**
+     * add RRDS or KSDS record to VES and update index XRBA address which
+     * may be in VX0 or in VES KSIR.
+     */
```
Comment (jyganci) [inline src/vz390.java:2455]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 36]
```diff
@@ -2454,13 +2635,16 @@ private void svc_rpl_point() {
...
-		 */
+
+
+    /**
+     * store RPLFEEDB 4 byte field with: 0 - pdf_ Problem Determination
+     * Field 1 - rc_ return code (also stored in R15) 2 - cmp_ component
+     * code 3 - rn_ reason code for corresponding rc_
+     * 
+     */
+    private void set_feedback(byte pdf, byte rc, byte cmp, byte rn) {
```
Comment (jyganci) [inline src/vz390.java:2646]: Add @param values to javadoc.

File: src/vz390.java [hunk 37]
```diff
@@ -2507,12 +2691,15 @@ private void set_feedback(byte pdf, byte rc, byte cmp, byte rn) {
...
-		 * VES ESDS base cluster data file at specified xrba. Notes: 1. If ESDS,
-		 * skip duplicate length after record. // RPI 672
-		 */
+
+
+    /**
+     * read record from VES into RPLAREA and set length in RPLLREC to the
+     * VES ESDS base cluster data file at specified xrba. Notes: 1. If ESDS,
+     * skip duplicate length after record. // RPI 672
+     */
```
Comment (jyganci) [inline src/vz390.java:2700]: Add @return to javadoc.

File: src/vz390.java [hunk 38]
```diff
@@ -2653,15 +2840,21 @@ private boolean read_ves_rec() {
...
+     * write current RPL record in RPLAREA with length RPLLREC to the VES
+     * base cluster data file at specified xrba.
+     * Notes:
+     * <ol>
+     * <li> Variable length records have 4 byte length preceeding record. </li>
+     * <li> ESDS variable length records also have 4 byte length following record
+     *      to support BWD read backward option without any index. </li>
+     * <li> Used by ESDS, RRDS, and KSDS when not inserted record </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:2855]: Add @return to javadoc.

File: src/vz390.java [hunk 39]
```diff
@@ -2747,10 +2940,13 @@ private boolean write_ves_rec() {
...
 
-	private boolean write_ves_cb(long cb_xrba, int cb_len) {
-		/*
-		 * write control block in ves at xrba into cb byte array Notes:
-		 */
+
+
+    /**
+     * write control block in ves at xrba into cb byte array Notes:
+     */
```
Comment (jyganci) [inline src/vz390.java:2947]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 40]
```diff
@@ -2778,10 +2974,13 @@ private boolean write_ves_cb(long cb_xrba, int cb_len) {
...
 
-	private boolean read_ves_cb(long cb_xrba, int cb_len) {
-		/*
-		 * read control block in ves at xrba into cb byte array Notes:
-		 */
+
+
+    /**
+     * read control block in ves at xrba into cb byte array Notes:
+     */
```
Comment (jyganci) [inline src/vz390.java:2981]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 41]
```diff
@@ -2820,12 +3019,15 @@ private boolean read_ves_cb(long cb_xrba, int cb_len) {
...
-		 * and cur_key
-		 * 
-		 */
+
+
+    /**
+     * read ksds index entry with key at cur_vx0_xrba and set cur_ves_xrba
+     * and cur_key
+     * 
+     */
```
Comment (jyganci) [inline src/vz390.java:3028]: Add @return to javadoc.

File: src/vz390.java [hunk 42]
```diff
@@ -2869,11 +3071,14 @@ private boolean read_ksds_index() {
...
-		/*
-		 * read ksir key at cur_ksir_rec + cur_vclr_koff into cur_key
-		 * 
-		 */
+
+
+    /**
+     * read ksir key at cur_ksir_rec + cur_vclr_koff into cur_key
+     * 
+     */
```
Comment (jyganci) [inline src/vz390.java:3079]: Add @return to javadoc.

File: src/vz390.java [hunk 43]
```diff
@@ -2918,11 +3123,14 @@ private boolean read_ksir_key() {
...
-		/*
-		 * read ves xrba prt in vx0 at cur_vx0_xrba and set cur_ves_xrba
-		 * 
-		 */
+
+
+    /**
+     * read ves xrba prt in vx0 at cur_vx0_xrba and set cur_ves_xrba
+     * 
+     */
```
Comment (jyganci) [inline src/vz390.java:3131]: Add @return to javadoc.

File: src/vz390.java [hunk 44]
```diff
@@ -2954,10 +3162,13 @@ private boolean read_xrba_ptr() {
...
 
-	private boolean write_ksds_index() {
-		/*
-		 * write ksds primary index entry with last_ves_xrba ptr and key.
-		 */
+
+
+    /**
+     * write ksds primary index entry with last_ves_xrba ptr and key.
+     */
```
Comment (jyganci) [inline src/vz390.java:3169]: Add @return to javadoc.

File: src/vz390.java [hunk 45]
```diff
@@ -2993,12 +3204,14 @@ private boolean write_ksds_index() {
...
-		/*
-		 * write ves rcd xrba ptr at vx0/ves(KSIR) index xrba in and update
-		 * cache for reuse
-		 */
+
+
+    /**
+     * write ves rcd xrba ptr at vx0/ves(KSIR) index xrba in and update
+     * cache for reuse
+     */
```
Comment (jyganci) [inline src/vz390.java:3212]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 46]
```diff
@@ -3025,11 +3238,14 @@ private boolean write_xrba_ptr(int tiot_index, long xrba_index,
...
-		/*
-		 * reset file length and tiot_eof addr for files being resused either
-		 * due to reuse option or REPRO seq out options..
-		 */
+
+
+    /**
+     * reset file length and tiot_eof addr for files being resused either
+     * due to reuse option or REPRO seq out options..
+     */
```
Comment (jyganci) [inline src/vz390.java:3246]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 47]
```diff
@@ -3041,11 +3257,14 @@ private boolean reuse_file(int adcb) {
...
-		/*
-		 * backup to next record for ESDS SEQ BWD retrieval and return logical
-		 * EOD error if at front of file.
-		 */
+
+
+    /**
+     * backup to next record for ESDS SEQ BWD retrieval and return logical
+     * EOD error if at front of file.
+     */
```
Comment (jyganci) [inline src/vz390.java:3265]: Add @return to javadoc.

File: src/vz390.java [hunk 48]
```diff
@@ -3109,12 +3328,18 @@ private boolean set_esds_bwd_next() {
...
+
+
+    /**
+     * compare key in RPLAREA+KEYOFF with key in last_vx0_key or cur_vx0_key
+     * array using comp_key_len set by rpl fetch -1, 0, 1 for low, equal, high
+     * Notes:
+     * <ol>
+     * <li> Set matching_key_bytes for generic key processing </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:3340]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 49]
```diff
@@ -3146,11 +3371,14 @@ private int comp_key(int key1_loc, byte[] key_byte) {
...
-		/*
-		 * compare key in RPLAREA+KEYOFF with high values RPI 779 using
-		 * cur_vclr_klen -1, 0, 1 for low, equal, high.
-		 */
+
+
+    /**
+     * compare key in RPLAREA+KEYOFF with high values RPI 779 using
+     * cur_vclr_klen -1, 0, 1 for low, equal, high.
+     */
```
Comment (jyganci) [inline src/vz390.java:3379]: Add @return to javadoc.

File: src/vz390.java [hunk 50]
```diff
@@ -3164,15 +3392,19 @@ private boolean rpl_key_high_values() {
...
+
+    /**
+     * get VSAM Cache Buffer (VCB) for file tiot_index, xrba, rec_len 
+     * <ol>
+     * <li> If rec_len > max_vcb_lrec return false. </li>
+     * <li> search for allocated vcb If not found add new vcb up to  </li>
+     *      max_vcb else replace least recently used allocated vcb. 
+     * <li> Set vcb_index and return true. </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:3405]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 51]
```diff
@@ -3239,12 +3474,15 @@ private void update_vcb_mru() {
...
-		 * 
-		 * if found set vcb_index and return true else false
-		 */
+
+
+    /**
+     * find matching vcb with same tiot, xrba, and record length
+     * 
+     * if found set vcb_index and return true else false
+     */
```
Comment (jyganci) [inline src/vz390.java:3483]: Add @return to javadoc.

File: src/vz390.java [hunk 52]
```diff
@@ -3262,12 +3500,17 @@ private boolean find_vcb() {
...
+
+
+/**
+ * calc vcb hash index from tiot + xrba + lrec
+ * Notes:
+ * <ol>
+ * <li> If two vcb requests geenrate same hash only the latest will be found and the
+ *      prior one will eventually be replaced via lru. </li>
+ * </ol>
+ */
```
Comment (jyganci) [inline src/vz390.java:3512]: Add @return to javadoc.

File: src/vz390.java [hunk 53]
```diff
@@ -3281,10 +3524,13 @@ private int calc_vcb_hash() {
...
 
-	private boolean read_ksit() {
-		/*
-		 * read KSIT into cb array and set cur_ksit xrbas
-		 */
+
+
+    /**
+     * read KSIT into cb array and set cur_ksit xrbas
+     */
```
Comment (jyganci) [inline src/vz390.java:3531]: Add @return to javadoc.

File: src/vz390.java [hunk 54]
```diff
@@ -3303,10 +3549,13 @@ private boolean read_ksit() {
...
 
-	private boolean read_ksir(long xrba) {
-		/*
-		 * read KSIR int cb and set cur_ksit xrba's
-		 */
+
+
+    /**
+     * read KSIR int cb and set cur_ksit xrba's
+     */
```
Comment (jyganci) [inline src/vz390.java:3556]: Add @return to javadoc.

File: src/vz390.java [hunk 55]
```diff
@@ -3336,10 +3585,13 @@ private boolean read_ksir(long xrba) {
...
 
-	private boolean update_ksds_rec() {
-		/*
-		 * rewrite ksds record following successful GET with UPD
-		 */
+
+
+    /**
+     * rewrite ksds record following successful GET with UPD
+     */
```
Comment (jyganci) [inline src/vz390.java:3592]: Add @return to javadoc.

File: src/vz390.java [hunk 56]
```diff
@@ -3366,10 +3618,13 @@ private boolean update_ksds_rec() {
...
 
-	private boolean insert_ksds_rec() {
-		/*
-		 * insert ksds record following unsuccessful GET with UPD
-		 */
+
+
+    /**
+     * insert ksds record following unsuccessful GET with UPD
+     */
```
Comment (jyganci) [inline src/vz390.java:3625]: Add @return to javadoc.

File: src/vz390.java [hunk 57]
```diff
@@ -3390,11 +3645,14 @@ private boolean insert_ksds_rec() {
...
-		/*
-		 * insert new KSIT for cur_vx0 entry with KSIR for existing record and
-		 * KSIR for new inserted record
-		 */
+
+
+    /**
+     * insert new KSIT for cur_vx0 entry with KSIR for existing record and
+     * KSIR for new inserted record
+     */
```
Comment (jyganci) [inline src/vz390.java:3653]: Add @return to javadoc.

File: src/vz390.java [hunk 58]
```diff
@@ -3490,21 +3748,28 @@ private boolean insert_ksit() {
...
-	private long alloc_ves(int cb_len) {
-		/*
-		 * return xrba of new ves cb/rec area and update eof xrba
-		 */
+
+
+
+    /**
+     * return xrba of new ves cb/rec area and update eof xrba
+     */
```
Comment (jyganci) [inline src/vz390.java:3756]: Add @return to javadoc.

File: src/vz390.java [hunk 59]
```diff
@@ -3490,21 +3748,28 @@ private boolean insert_ksit() {
...
-		 * at current KSIR and balance AVT
-		 * tree by rotating KSIR's if needed.
-		 */
+
+
+    /**
+     * insert new KSIR to existing KSIT 
+     * at current KSIR and balance AVT
+     * tree by rotating KSIR's if needed.
+     */
```
Comment (jyganci) [inline src/vz390.java:3770]: Add @return to javadoc.

File: src/vz390.java [hunk 60]
```diff
@@ -3634,20 +3899,24 @@ private boolean insert_ksir() {
...
+     * <li> Increment height of all nodes on path prior to unbalanced node. </li>
+     * <li> Find first (and only) node which may be unbalanced
+     *      (left vs right height differs by 2). </li>
+     * <li> If unbalanced node found, set avl_unbalanced and avl_r_xrba </li>
+     * </ol>
+     * Notes:
+     * <ol>
+     * <li> Only returns false if I/O error </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:3917]: Add @return to javadoc.

File: src/vz390.java [hunk 61]
```diff
@@ -3731,11 +4000,13 @@ private boolean avl_update_height(){
...
-    	/*
-    	 * verify updated T1-T4 within +-1
-    	 */
+
+
+
+    /**
+     * verify updated T1-T4 within +-1
+     */
+    private boolean check_heights(String type, byte t1, byte t2, byte t3, byte t4){
```
Comment (jyganci) [inline src/vz390.java:4009]: Add @param and @return values to javadoc.

File: src/vz390.java [hunk 62]
```diff
@@ -3747,48 +4018,59 @@ private boolean check_heights(String type,
     	}
     	 return true;  
     }
+
+
+
+    /**
+     * display unbalanced KSIR found
+     * 
+     */
```
Comment (jyganci) [inline src/vz390.java:4027]: Add @param to javadoc.

File: src/vz390.java [hunk 63]
```diff
@@ -3747,48 +4018,59 @@ private boolean check_heights(String type,
...
+     * <li> If node found with new +2 state, rotate as follows
+     *      to rebalance which ends the update process.
+     *      <ol>
+     *      <li> LL - move left left node up one level by swapping n ode with left node. </li>
+     *      <li> RR - move right right node up one level by swapping node with right node </li>
+     *      <li> LR - move left right node up one level by swapping node with left right node </li>
+     *      <li> RL - move right left node up one level by swapping mode with right left node. </li>
+     *      </ol> </li>
+     * </ol>
+     */
```
Comment (jyganci) [inline src/vz390.java:4071]: Add @return to javadoc.

File: src/vz390.java [hunk 64]
```diff
@@ -3847,11 +4129,15 @@ private boolean avl_rotate_ksir() {
 		cur_ksir_xrba = save_ksir_xrba; // restore to inserted ksir
 		return true;
 	}
+
+
+
+    /**
+     * avl rotate x to r, and r to x_high,
+     * and x_high to r_low. 
+     */
```
Comment (jyganci) [inline src/vz390.java:4138]: Add @return to javadoc.

File: src/vz390.java [hunk 65]
```diff
@@ -3927,12 +4213,16 @@ private boolean avl_rotate_left_left(){
...
     	return true;
     }
+
+
+
+    /**
+     * avl rotate w (x_high) to r,
+     * r to w_high, w_low to x_high,
+     * and w_high to r_low.
+     */
```
Comment (jyganci) [inline src/vz390.java:4223]: Add @return to javadoc.

File: src/vz390.java [hunk 66]
```diff
@@ -4041,11 +4331,15 @@ private boolean avl_rotate_left_right(){
     	}
     	return true;
     }
+
+
+
+    /**
+     * avl rotate x to r, and r to x_low,
+     * and x_low to r_high.
+     */
```
Comment (jyganci) [inline src/vz390.java:4340]: Add @return to javadoc.

File: src/vz390.java [hunk 67]
```diff
@@ -4121,12 +4415,16 @@ private boolean avl_rotate_right_right(){
...
     	return true;
     }
+
+
+
+    /**
+     * avl rotate w (x_low) to r,
+     * r to w_low, w_high to x_low,
+     * and w_low to r_high.
+     */
```
Comment (jyganci) [inline src/vz390.java:4425]: Add @return to javadoc.

File: src/vz390.java [hunk 68]
```diff
@@ -4235,10 +4533,14 @@ private boolean avl_rotate_right_left(){
...
-	private boolean write_ksit() {
-		/*
-		 * write new KSIT at ves eof xrba
-		 */
+
+
+
+    /**
+     * write new KSIT at ves eof xrba
+     */
```
Comment (jyganci) [inline src/vz390.java:4541]: Add @return to javadoc.

File: src/vz390.java [hunk 69]
```diff
@@ -4254,10 +4556,12 @@ private boolean write_ksit() {
...
 
-	private boolean write_ksir() {
-		/*
-		 * write ves KSIR at cur_ksir_xrba
-		 */
+
+
+    /**
+     * write ves KSIR at cur_ksir_xrba
+     */
```
Comment (jyganci) [inline src/vz390.java:4563]: Add @return to javadoc.

File: src/tz390.java (hunk 10 above)
Comment (abekornelis) [reply src/tz390.java:6997]: The routine refers t dfp_digits where I think it should refer to tot_digits. Will work okay as long as global variable dfp_digits is indeed intended to be the input field. Do we need to change this? Could you please have a look at it?

**Pull request closed without merging**